### PR TITLE
Heavy Widget construction and destruction performance test

### LINKED ISSUE #60300: A performance test case for heavy Widget building
Currently we don't have a test case that have heavy Widget constructions and destructions: yes it's something we want to avoid in a flutter app, but such test case can be helpful when: 
1. Measuring the overhead of a test binding: such test case can make sure that overhead from Widget and element rebuilding to Rendering Object repainting are all covered. 
1. Measuring the worse frame performance. 99th percentile number and worst frame build time is our current metric for this but they are noisy without multiple runs and averaging.

### PR DESCRIPTION
**Pull request by CareF**
## Description

This add a stress test for the performance of constructing and destructing Widgets. 

## Related Issues

Closes #60300

## Tests

I added the following tests:

`dev/benchmarks/macrobenchmarks/test_driver/multi_widget_construction_perf.dart`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/devicelab/manifest.yaml [hunk 1]
```diff
@@ -156,6 +156,12 @@ tasks:
     stage: devicelab
     required_agent_capabilities: ["mac/android"]
 
+  multi_widget_construction_perf__timeline_summary:
+    description: >
+      Measures the runtime performance of constructing and destructing widgets on Android.
+    stage: devicelab
+    required_agent_capabilities: ["mac/android"]
```
Comment (liyuqian) [inline dev/devicelab/manifest.yaml:163]: Nit: the infra team now prefers to use `linux/android`

**Review by liyuqian — APPROVED** [review]
LGTM with a nit. Remember to baseline https://flutter-dashboard.appspot.com/benchmarks.html once this test have enough data collected :)

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by CareF** [comment]
I'm not sure if I should ignore this, but `Cirrus CI / docs-linux` keeps reporting timeout. Re-run doesn't fix. @liyuqian

**Comment by flar** [comment]
There was a glaring error in the way this benchmark was introduced that wasn't caught during code reviews. You should add a new button to launch your benchmark, not replace an existing button with yours. I'll add the old button back in rather than revert this as it will get us to the end state quicker, but be careful about that in the future.

File: dev/benchmarks/macrobenchmarks/lib/main.dart [hunk 2]
```diff
@@ -136,10 +138,10 @@ class HomePage extends StatelessWidget {
             },
           ),
           RaisedButton(
-            key: const Key(kImageFilteredTransformAnimationRouteName),
```
Comment (flar) [inline dev/benchmarks/macrobenchmarks/lib/main.dart:139]: Adding a benchmark should never involve deleting code so this diff should have been a red flag during the reviews...

Comment (CareF) [reply dev/benchmarks/macrobenchmarks/lib/main.dart:139]: This is not intended. Some how when I copy paste things I override it :( I added it to a new patch #60412

Comment (flar) [reply dev/benchmarks/macrobenchmarks/lib/main.dart:139]: I've already fixed this in https://github.com/flutter/flutter/commit/2962912d68fb72d6c9fd60f652e27f3593d692d6

It has already been pushed to the main tree.

**Pull request merged**